org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/JsonRpcException.java

```diff
@@ -27,6 +27,7 @@ public class JsonRpcException extends RuntimeException {
 	public static boolean indicatesStreamClosed(Throwable thr) {
 		return thr instanceof InterruptedIOException
 				|| thr instanceof ClosedChannelException
+				|| thr instanceof IOException && "Stream closed".equals(thr.getMessage())
 				|| thr instanceof IOException && "Pipe closed".equals(thr.getMessage())
```

Comment:
Thanks for the patch. LGTM. I would only suggest

```suggestion
				|| thr instanceof IOException &&
					("Stream closed".equals(thr.getMessage()) ||
					 "Pipe closed".equals(thr.getMessage()))
```

mostly for the sake of consistency with the `SocketException` case.